OpenRA.Mods.Common/Traits/SpawnActorsOnSell.cs

```diff
@@ -72,6 +76,33 @@ void Emit(Actor self)
 			}
 
 			var eligibleLocations = buildingInfo.Tiles(self.Location).ToList();
+
+			var guaranteedActorTypes = Info.GuaranteedActorTypes?.Select(a =>
+			{
+				var av = self.World.Map.Rules.Actors[a].TraitInfoOrDefault<ValuedInfo>();
```

Comment:
Would a null check be wise? Will Actors[a] always return something. Will the yaml have been validated to only contain correctly spelled actor types?

I.e. to avoid bugs showing up only at the first sell.

Reply:
We have [ActorReference] attributes to test for invalid actor names